accloudtant/aws/prices.py

```diff
@@ -108,17 +115,79 @@ def print_prices(instances=None, region='us-east-1'):
             hourly_price = rsrvd_3yr[all_upfront]['effectiveHourly']
             all_upfront_3yr = eval_price_exists(hourly_price)
-            row = [size,
+            row = [region,
+                   size,
                    on_demand,
                    no_upfront,
                    partial_upfront_1yr,
                    all_upfront_1yr,
                    partial_upfront_3yr,
                    all_upfront_3yr]
             table.append(row)
-    output = "EC2 (Hourly prices, no upfronts, no instance type features):\n"
-    output += tabulate(table, headers)
-    return output
+        if output_format == 'table':
+            output = "EC2 (Hourly prices, no upfronts, no instance type features):\n"
+            output += tabulate(table, headers,tablefmt="fancy_grid")
+            if save != 'no':
+                print ('Please note table format will not be saved in any files')
+            return output
+
+        elif output_format == 'csv':
+            output = ''
+            for header in headers:
+                output += header + ','
+            output = output[:-1] + '\n'
+            for row in table:
+                for column in row:
+                    output += str(column) + ','
+                output = output[:-1] + '\n'
+            if save != 'no':
+                output_file_name = save + "/accloudtant-list-" + str(time_now) + "-" + region + ".csv"
+                output_file = open(output_file_name,'w')
+                output_file.write(output[:-1])
+                output_file.close()
+                print ('The output is saved in ' + output_file_name)
+            return output[:-1]
+
+        elif output_format == 'json':
+            json_output = "[\n{\n"
+            region = table[0][0]
+            json_output += '"' + region + '":{\n' + '"Instances":[\n{\n'
+            number_rows = len(table)
+            number_columns = len(table[0])
+            row = 0
+            last_column = number_columns - 1
+            last_row = number_rows - 1
+            while row < number_rows:
+                col = 1
+                while col < number_columns:
+                    if col == 1:
+                        json_output += '"' + str(table[row][col]) + '":{\n'
+                        col += 1
+                        continue
+                    else:
+                        json_output += '"' + headers[col] + '" : '
+                    if col == last_column:
+                        json_output += '"' + str(table[row][col]) + '"\n}'
+                    else:
+                        json_output += '"' + str(table[row][col]) + '",\n'
+                    col += 1
+                if row == last_row:
+                    json_output += '\n}\n]\n}\n}\n]'
+                else:
+                    json_output += ",\n"
+                row += 1
+            json_output = json.loads(json_output)
+            json_pretty = json.dumps(json_output,indent=4, sort_keys=True)
```

Comment:
I don't see the need for pretty json formatting. This can be done using `jq` ([1](https://stedolan.github.io/jq/)), emmbracing UNIX command philosophy.